### PR TITLE
RT-028 Extend button by loader

### PR DESCRIPTION
**Pull request by dominikszczuka**

### CODE REVIEW

File: src/Components/CustomButton/CustomButton.tsx [hunk 1]
```diff
@@ -19,7 +23,10 @@ const CustomButton: React.FC<CustomButtonProps> = ({
   customStyles,
   filled,
   icon,
+  isLoading,
 }) => {
+  const { loading } = useSelector((state: AppState) => state.todosReducer);
```
Comment (MRadek55) [inline src/Components/CustomButton/CustomButton.tsx]: Why are you using two different loading props ?

File: src/Components/CustomButton/CustomButton.tsx [hunk 2]
```diff
@@ -33,8 +40,11 @@ const CustomButton: React.FC<CustomButtonProps> = ({
         )}
         data-filled={filled}
       >
-        {label}
-        {!!icon && <Icon {...icon} />}
+        {isLoading && loading ? (
```
Comment (MRadek55) [inline src/Components/CustomButton/CustomButton.tsx]: Why like that ?

File: src/Components/CustomButton/CustomButton.tsx [hunk 3]
```diff
@@ -33,8 +40,11 @@ const CustomButton: React.FC<CustomButtonProps> = ({
...
+          <Loader />
+        ) : (
+          label && !!icon && <Icon {...icon} />
```
Comment (MRadek55) [inline src/Components/CustomButton/CustomButton.tsx]: Why icon is dependent to label ?

File: src/Screens/HomeScreen/HomeScreen.tsx [hunk 4]
```diff
@@ -31,6 +31,7 @@ const HomeScreen = () => {
         color: `${palette.white}`,
       }}
       customStyles={styles.customBtnStyles}
+      isLoading={true}
```
Comment (MRadek55) [inline src/Screens/HomeScreen/HomeScreen.tsx]: Here should be your loading from redux

**Review by WiktorMonka — APPROVED** [review]

File: src/Components/CustomButton/CustomButton.tsx [hunk 5]
```diff
@@ -36,15 +39,12 @@ const CustomButton: React.FC<CustomButtonProps> = ({
           styles.btn,
           typography.buttonFont,
           styles.hover,
-          customStyles
+          customStyles,
+          disabled ? styles.disabled : null
```
Comment (MRadek55) [inline src/Components/CustomButton/CustomButton.tsx]: `disabled && styles.disabled` should be enough

**Review by MRadek55 — APPROVED** [review]

**Pull request merged**